docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,310 @@
+[[fleet-agent-proxy-support]]
+= Use a proxy server with {agent} and {fleet}
+
+Your organization’s security strategy and other considerations may require you
+to use a proxy server between some components in your deployment. For example,
+you may have a firewall rule that prevents endpoints from connecting directly to
+{es}. In this scenario, you can set up the {agent} to connect to a proxy, then
+the proxy can connect to {es} through the firewall.
+
+Support is available in {agent} and {fleet} for connections through HTTP Connect
+(HTTP 1 only) and Socks5 proxy servers.
+
+This guide describes where a proxy server is allowed in your deployment and how
+to configure proxy settings for {agent} and {fleet}. The steps for deploying the
+proxy server itself are beyond the scope of this article.
+
+NOTE: Some environments require users to authenticate with the proxy. There are
+no explicit settings for proxy authentication in {agent} or {fleet}, except the
+ability to pass credentials in the URL or as keys/tokens in headers, as
+described later.
+
+[discrete]
+[[elastic-agent-proxy-config]]
+== Configure {agent} proxy settings
+
+Configure proxy settings for {agent} when it must connect through a proxy server
+to:
+
+* Download artifacts from `artifacts.elastic.co` for subprocesses or binary
+upgrades
+* Send data to {es}
+* Retrieve agent policies from {fleet-server}
+* Retrieve agent policies from {es} (only needed for agents running {fleet-server})
+* Monitor services (not shown)
+
+image::images/agent-proxy-server.png[Image showing connections between {agent}, {fleet-server}, and {es}]
+
+You can configure {agent} proxy settings by specifying:
+
+* <<host-proxy-env-vars>>
```

Comment:
Can we say why a user would choose one over the other? Eg. Env vars are good when you want to proxy everything on the agent or even across the whole host. The agent policy really only lets you control the proxy to the ES output (I'm proposing we remove the other cases below). The CLI commands let you override environment variables set at the system level and specify a different proxy for agent. @michel-laterman does that sound correct?

Reply:
Nevermind, it looks like these scenarios are described below.